emcc.py

```diff
@@ -1741,7 +1733,7 @@ def phase_linker_setup(options, state, newargs, user_settings):
     # libraries because STACK_OVERFLOW_CHECK depends on emscripten_stack_get_end which is defined
     # in libcompiler-rt.
     if not settings.PURE_WASI and '-nostdlib' not in newargs and '-nodefaultlibs' not in newargs:
-      default_setting(user_settings, 'STACK_OVERFLOW_CHECK', max(settings.ASSERTIONS, settings.STACK_OVERFLOW_CHECK))
+      default_setting(user_settings, 'STACK_OVERFLOW_CHECK', max(int(settings.ASSERTIONS), settings.STACK_OVERFLOW_CHECK))
```

Comment:
why do we need to cast to int here? isn't that setting always an integer, before and after this PR?

Reply:
Ah good point.. this must have been from some intermediate point in my work.